src/java.base/share/classes/java/io/FileInputStream.java

```diff
@@ -354,6 +354,29 @@ public int read(byte b[], int off, int len) throws IOException {
         return (capacity == nread) ? buf : Arrays.copyOf(buf, nread);
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    public long transferTo(OutputStream out) throws IOException {
+        if (out instanceof FileOutputStream fos) {
+            FileChannel fci = getChannel();
+            long pos = fci.position();
+            long count = fci.size() - pos;
```

Comment:
It might be better to just specify the count as Long.MAX_VALUE and test the size after the transfer. This would be a bit more robust in the scenario that the file grows and would avoid fallback when the file is truncated, e.g.

FileChannel fc = in.getChannel();
long pos = fc.position();
long transferred = fc.transferTo(pos, Long.MAX_VALUE, out.getChannel());
long newPos = pos + transferred;
fc.position(newPos);
if (newPos >= fc.size()) {
    return transferred;
}